sdk/src/main/java/com/google/cloud/dataflow/sdk/io/OffsetBasedSource.java

```diff
@@ -264,7 +264,18 @@ public final boolean start() throws IOException {
 
     @Override
     public final boolean advance() throws IOException {
-      return advanceImpl() && rangeTracker.tryReturnRecordAt(isAtSplitPoint(), getCurrentOffset());
+      if (!advanceImpl()) {
+        // Reader could not read another record. Add an extra split point in order to "finish" the
```

Comment:
I'd prefer not to reuse tryReturnRecordAt for this purpose - better add a method that specifically tells the range tracker that EOF was reached. This is not an extra split point because split points are existing records at split point positions, and there is no record here. We're just telling the range tracker that the *previous* split point (and its block) has been processed so it's okay to increment the counter.